testing/trino-tests/src/test/java/io/trino/execution/TestUserImpersonationAccessControl.java

```diff
@@ -95,14 +95,14 @@ private QueryError trySelectQuery(String assumedUser)
                     .build();
 
             // start query
-            StatementClient client = newStatementClient(httpClient, clientSession, "SELECT * FROM tpch.tiny.nation");
+            try (StatementClient client = newStatementClient(httpClient, clientSession, "SELECT * FROM tpch.tiny.nation")) {
```

Comment:
nit: look unrelated - separate commit?

Reply:
If the statement is not closed the query never finishes and doesn't receive a final TaskInfo making the newly introduced check fail

Reply:
Still should be a preparatory commit. So adding checks and making project actually pass the checks are separate commits. (fixing makes sense even if we do not have checks).